### DISCUSSION TITLE
Using setup wizard to set up limit switches causes "alarm" state

### DISCUSSION THREAD

**Discussion by noweare1** [winder/Universal-G-Code-Sender#2472] [General]
Using setup wizard to configure the machine. When I manually trigger any limit switch I get the error msg "An unexpected error was detected: ALARM: Hard limit [Reset to continue].  I end up having to exit the setup wizard, unlocking and doing a soft reset to clear the "alarm".  I would like to know if this is the correct way to set up limit switches and if this is abnormal operation.

**Comment by breiler** [comment]
Just to make sure, which step are you trying to configure?

What firmware/version are you using?

**Reply by noweare1** [reply]
Universal GCode Sender version 2.1.5

step 5 : limit switches (using setup wizard.) 
Thank you.

**Reply by breiler** [reply]
What GRBL version are you using?

**Reply by noweare1** [reply]
0.9 j  I thought I was using v1.1 kind of confusing as they are changing where they are hosting the files.

**Reply by noweare1** [reply]
The limit switches are working. The UGS throws an "alarm" but if I hold the limit switches down for longer (like the text on screen told me to do)  I see the color changes from green to red. Thanks for your help.

**Reply by breiler** [reply]
UGS does not throw any alarms, GRBL does. The Alarm state is normal and is the only way to be able to test the limit switches. A more user friendly way would be to filter out any console messages and not show the current state of the controller in DRO window in the background, but that is currently not possible.

You should not need to close the Setup Wizard, it should be able to take care of all those alarm states for you.
